### DISCUSSION TITLE
excessive query diffs after build

### DISCUSSION THREAD

**Discussion by joshbooker** [joyfullservice/msaccess-vcs-addin#414] [General]
Hi,

Is there a known issue with excessive query diffs after build?  

When I do this flow, I should expect no changes, correct?

Export > Commit > Build > Export 

Instead, get the same 100+ queries every time after build.

It appears the discrepancy is in the `JOIN` clause reordering the join fields.
![image](https://github.com/joyfullservice/msaccess-vcs-addin/assets/2698896/cb4a11b6-8ad4-4de7-946f-88e03c52d3b5)

I believe I've seen similar reordering of fields on the `WHERE` clause as well but can't find the example ATM.

TIA

**Comment by joyfullservice** [comment]
In my experience, the _first_ time you build a database from source, you will typically see some "noise" when you export the new build. I usually just commit those changes as "Build from source". Subsequent builds of the same project should (ideally) not create any new changes in git. If you build from source, tweak a form, export, build from source again, the source file from the form should be the only thing showing as a change in the repository.

That being said, queries can be a bit of an interesting component. If you edit the query in the SQL view, it is saved one way, whereas if you save it through the designer, it is saved another way. You also have factors like Access optimizing queries, and some rare issues with specific join scenarios. You can find some discussions on these in previous issues.

- #76 
- #219 
- #391 

You also have an option to [force the import of the SQL during build](https://github.com/joyfullservice/msaccess-vcs-addin/wiki/Options#build), which can help in some edge case scenarios. I would recommend reading through the issues mentioned above first, as those contain some helpful tips for understanding VCS changes in query objects.

Hope that helps!
